host_core/lib/host_core/providers/provider_supervisor.ex

```diff
@@ -34,7 +34,17 @@ defmodule HostCore.Providers.ProviderSupervisor do
     [{pid, _val}] = Registry.lookup(Registry.ProviderRegistry, {public_key, link_name})
     Logger.info("About to terminate child process")
     prefix = HostCore.Host.lattice_prefix()
-    :ok = Gnat.pub(:lattice_nats, "wasmbus.rpc.#{prefix}.#{public_key}.#{link_name}.shutdown", "")
+    # Allow provider 2 seconds to clean up resources
+    case Gnat.request(
```

Comment:
Have the providers been modified to respond to this as a request? It used to be an async publish

Reply:
Just modified the two providers we have, and I added a Makefile so we can easily rebuild them. It's not a perfect makefile but it'll do for now, especially when I have to swap them out for the Linux executables for committing.

Feasibly, we could have that part automated as well.